File: java/Primes.java
/***********************************************************
    primes.java -- 素数（Java）
    javac Primes.java
    java Primes
***********************************************************/
public class Primes {
    static int[] generatePrimes(int total) {
        int primes[] = new int[total];
        primes[0] = 2;
        int k = 1;
        for (int x = 3; k < total; x += 2) {
            int i = 0;
            while (i < k && x % primes[i] != 0) {
                if (primes[i] * primes[i] >= x) {
                    primes[k++] = x;
                    break;
                }
                i++;
            }
        }
        return primes;
    }

    public static void main(String[] args) {
        int total = 10000;
        System.out.println("素数表 Java");

        long start_time = System.nanoTime();
        int[] primes = generatePrimes(total);
        long end_time = System.nanoTime();
        System.out.println();
        System.out.println("計算時間：" + (double)(end_time - start_time)/1000000000  + "秒");
        start_time = System.nanoTime();
        for (int i = 0; i < total; i++) {
            if (i % 10 == 0) System.out.println();
            System.out.print(String.format("%8d",primes[i]));
        }
        end_time = System.nanoTime();

        System.out.println();
        System.out.println("出力時間：" + (double)(end_time - start_time)/1000000000  + "秒");
    }

}

File: go/primes.go
/***********************************************************
    primes.go -- 素数（Go言語）
    go run primes.go
***********************************************************/
package main
import (
    "fmt"
    "time"
)

func generate_primes(total int) []int {
    var primes [] int
    primes = append(primes, 2)
    var x, j, k int = 1, 0, 1
    for {
        if k >= total {
            break
        }
        x += 2
        j = 0
        for {
        	if j >= k || x % primes[j] == 0 {
                break
            }
            if primes[j] * primes[j] > x {
                k++
                primes = append(primes, x)
                break
            }
            j++
        }
    }
    return primes
}

func main() {
    fmt.Println("素数表 Go")
    var total int = 10000
    start_time := time.Now()
    var primes = generate_primes(total)
    end_time := time.Now()
    fmt.Println("")
    fmt.Printf("計算時間：%f秒\n", (end_time.Sub(start_time)).Seconds())

    start_time = time.Now()
    var i int = 0
    for i < total {
        if i % 10 == 0 {
            fmt.Println("")
        }
        fmt.Printf("%8d", primes[i])
        i++
    }
    end_time = time.Now()
    fmt.Println("")
    fmt.Printf("出力時間：%f秒\n", (end_time.Sub(start_time)).Seconds())
}